### ISSUE TITLE
Timeout for ModbusUdpClient

### ISSUE THREAD

**Issue by MAKOMO** [bashwork/pymodbus#51]
If the slave stops, the UDP client realized with pymodbus hangs (on OS X). There seems to be no way to specify a timeout on the underlying socket as for the ModbusSerialClient. The following would allow to use a timeout argument using the newly passed down kwargs.

```python
class ModbusUdpClient(BaseModbusClient):
    def __init__(self, host='127.0.0.1', port=Defaults.Port,
        framer=ModbusSocketFramer, **kwargs):
        ''' Initialize a client instance

        :param host: The host to connect to (default 127.0.0.1)
        :param port: The modbus port to connect to (default 502)
        :param framer: The modbus framer to use (default ModbusSocketFramer)
        '''
        self.host = host
        self.port = port
        self.socket = None
        self.timeout  = kwargs.get('timeout',  Defaults.Timeout)
        BaseModbusClient.__init__(self, framer(ClientDecoder()),**kwargs)
    
    def connect(self):
        ''' Connect to the modbus tcp server

        :returns: True if connection succeeded, False otherwise
        '''
        if self.socket: return True
        try:
            family = ModbusUdpClient._get_address_family(self.host)
            self.socket = socket.socket(family, socket.SOCK_DGRAM)
            self.socket.settimeout(self.timeout)
        except socket.error, ex:
            _logger.error('Unable to create udp socket %s' % ex)
            self.close()
        return self.socket != None
```

An update on the python3 build would be nice too.

Thanks,
Marko

**Comment by bashwork** [comment]
Fixed with b13c89d32ade35e6121d5e11d7f089551b92da62

**Comment by Eventect** [comment]
I believe the modification above has not been committed properly. It should read, as MAKOMO suggested:

    self.socket.settimeout(self.timeout)

but it appears to have been commited as:

    self.settimeout(self.timeout)

which deliver a no method error in case the timeout option is passed.
